### PR TITLE
Component testing setup

### PR DESCRIPTION
**Pull request by harshit-bs**
### Features 
- https://github.com/nightwatchjs/create-nightwatch/issues/72

### CODE REVIEW

**Comment by vishalshah133** [comment]
Hey, @beatfactor  Can we update the 1st question to single select with the following options?
End to End
Component
Both

Lesser things to read and lesser actions

**Comment by beatfactor** [comment]
> Hey, @beatfactor Can we update the 1st question to single select with the following options? End to End Component Both
> 
> Lesser things to read and lesser actions

I'm not sure. The next question will be a multi-select (browsers) so that's why I thought that we could keep the same design. We will also have API testing soon, so then we would have to change the design if we go for the single select.

File: assets/component-config/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import nightwatchPlugin from 'vite-plugin-nightwatch';
```
Comment (beatfactor) [inline assets/component-config/vite.config.js:1]: I don't think we need to generate a vite config file at this point. In most cases, there will already be one, but if not, each plugin has its own vite config file with everything needed.

File: src/constants.ts [hunk 2]
```diff
@@ -135,7 +176,15 @@ export const QUESTIONAIRRE: inquirer.QuestionCollection = [
     type: 'input',
     name: 'baseUrl',
     message: 'Enter the base_url of the project',
-    default: 'http://localhost'
+    default: (answers: {uiFramework: string}) => {
+      if (['react', 'vue'].includes(answers.uiFramework)) {
+        return 'localhost:5173';
```
Comment (beatfactor) [inline src/constants.ts:181]:
```suggestion
        return 'http://localhost:5173';
```

File: src/constants.ts [hunk 3]
```diff
@@ -135,7 +176,15 @@ export const QUESTIONAIRRE: inquirer.QuestionCollection = [
...
+      } else if (answers.uiFramework === 'storybook') {
```
Comment (beatfactor) [inline src/constants.ts:182]:
```suggestion
      }
      
      if (answers.uiFramework === 'storybook') {
```

File: src/init.ts [hunk 4]
```diff
@@ -133,6 +133,37 @@ export class NightwatchInit {
...
+      const componentConfigPath = path.join(__dirname, '..', 'assets', 'component-config');
+      const nightwatchPath = path.join(this.rootDir, 'nightwatch');
+
+      try {
+        fs.mkdirSync(nightwatchPath);
+        // eslint-disable-next-line
+      } catch (err) {}
+
+      if (answers.language === 'js') {
+        const viteSrcPath = path.join(componentConfigPath, 'vite.config.js');
```
Comment (beatfactor) [inline src/init.ts:148]: no need to add vite.config files

**Review by beatfactor — CHANGES_REQUESTED** [review]
I think we need a few more tests here.

**Pull request merged**